### DISCUSSION TITLE
Can't get to work Code coverage

### DISCUSSION THREAD

**Discussion by michael3m** [maziac/DeZog#66] [Q&A] [answered]
Hello.
I took a fresh install of VSCode inside VMware, installed DeZog and  Z80 Macro-Assembler from mborik on it - nothing more related to Z80 and Spectrum.
Then, I downloaded z80-sample-program and started debugging - and code coverage just won't shows up.
`            "history": {
                "reverseDebugInstructionCount": 1000000,
                "spotCount": 10,
                "codeCoverageEnabled": true
            }
`
is present in launch.lson.
What's the problem? Thanks in advance.

**Comment by maziac** [comment]
Could you maybe zip your project (together with the list file) and attach it here, so I could have a look.

**Reply by michael3m** [reply]
Actually, it is not my project, it is only z80-sample-program and nothing more.
However, Code coverage isnt working for me in my own projects aswell, but they all are based on launch.json that are coming with z80-sample-program.
But, anyway - here is the file.
[z80-sample-program-master.zip](https://github.com/maziac/DeZog/files/7073018/z80-sample-program-master.zip)

**Comment by maziac** [comment]
Ok, just tested (with your download) and for me it works:
![Bildschirmfoto 2021-08-29 um 22 41 02](https://user-images.githubusercontent.com/3081973/131264724-92a9a88e-9e5d-4021-a7f1-a6a233db0a97.jpg)

The (dark) green lines you see are the lines that have been covered.

I tested with the internal simulator and did a few steps:
![Bildschirmfoto 2021-08-29 um 22 42 18](https://user-images.githubusercontent.com/3081973/131264752-cd229442-609a-4952-8358-87f7a1029671.jpg)

Which Remote (emulator) have you been using and what OS you are on?

**Reply by michael3m** [reply]
![vmware_2021-08-30_00-42-10](https://user-images.githubusercontent.com/45968127/131266251-f259d3b8-fd32-45e6-8325-3d3602a11e50.png)
Well.
I used internal simulator aswell.
Tried on Windows 10 Pro.
I dont know what to say now. Got the same behavior on 2 PCs (they are syncing VSCode settings via Github) and 1 virtual PC (not syncing). 
Maybe, I am doing something wrong to see Code coverage at all? I just starting debug by pressing F5 and then press F10, F11 and "Step Back" buttons.
I even tried to change VSC color scheme, but no luck.

**Comment by maziac** [comment, marked as answer]
I found the problem: it's simply not working on Windows.
I already entered an error: #69
Will hopefully fix this soon.

**Comment by maziac** [comment]
I released a new version 2.3.2 which fixed this bug. Thanks for reporting.

It should be in the marketplace soon (I already uploaded).
If impatient you can also install it from here:
https://github.com/maziac/DeZog/releases/tag/v2.3.2

Please let me know if it works.

**Reply by michael3m** [reply]
Haha, really unexpected. 
Thanks for your work.
